core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/localscan/LocalCsvFileScanOpDesc.java

```diff
@@ -86,17 +82,102 @@ public Schema sourceSchema() {
     }
 
     private Schema inferSchema(String headerLine) {
+        java.util.List<AttributeType> attributeType = IntStream
+                .range(0, headerLine.split(delimiter).length).mapToObj(i->AttributeType.INTEGER).collect(Collectors.toList());
+
+
+        try(BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            if (header != null && header)
+                reader.readLine();
+            int i = 0;
+            while((line = reader.readLine())!=null && i<=100) {
+                String[] records = line.split((delimiter));
+                attributeType = inferline(attributeType, records);
+                i++;
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        java.util.List<AttributeType> finalAttributeType = attributeType;
```

Comment:
why do we make a reference copy?

Reply:
There is an error if I directly use the `attributeTypeList`. Only when I make a reference copy, I can get it working. Could you give me any suggestion on this?
<img width="467" alt="Screen Shot 2021-02-11 at 11 14 59 PM" src="https://user-images.githubusercontent.com/41406160/107740029-01d19d80-6cbf-11eb-8e83-745db06a7b91.png">